core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -405,8 +404,20 @@ public void remove() {
       }
     };
 
-    return CloseableIterator.noopCloseable(Iterators
-        .concat(CommonUtils.singleElementIterator(getContainerIdJournalEntry()), blockIterator));
+    CloseableIterator<JournalEntry> closeableIterator =
```

Comment:
looks to me a wrapper which only used when `blockStoreIterator` is `CloseableIterator` when using Rocksdb. Maybe it is a bit more cleaner and easier to maintain if we have the if switch earlier, e.g., 

```java
if (blockStoreIterator instanceof CloseableIterator) {
  // like your new code to concat with `closeableIterator`
} else {
  // keep the previous way to concat with `blockIterator` 
}
```

alternatively, can we just make `blockIterator` a `CloseableIterator`, will that be cleaner?

Reply:
I thought about this actually. The reason I prefer to do the switch in the closeAction lambda is, we will always try to close the iterator and only give up if there's nothing to close, instead of not trying at all by using `CloseableIterator.noopCloseable()`. Considering this iteration only happens at when we iterate the journal (backup or other kinds of dumping), the performance difference should be small. WDYT? @apc999